core/src/main/java/com/linecorp/armeria/server/DefaultRoute.java

```diff
@@ -209,8 +249,15 @@ public int complexity() {
         return produces;
     }
 
+    @Override
+    public String cacheKey() {
+        return cacheKey;
```

Comment:
I'm not sure if it's a good idea to have a `String` key like this. What do you think about replacing `cacheKey` with two member fields and their getters:

- `Set<String> predicateParamNames()`
- `Set<AsciiString> predicateHeaderNames()`

Then, we could calculate `noCacheRoutes` by comparing `Route`s' properties. Thoughts?

Reply:
Agreed. I removed cacheKey from Route. Now compare pathType, paths, methods, consumes and produces. Please have a look and let me know what you think about this change. :-)